File: app/core/src/main/java/com/fsck/k9/Identity.java
package com.fsck.k9;

import java.io.Serializable;

public class Identity implements Serializable {
    private static final long serialVersionUID = -1666669071480985760L;

    private String description;
    private String name;
    private String email;
    private String signature;
    private boolean signatureUse;
    private String replyTo;

    public synchronized String getName() {
        return name;
    }

    public synchronized void setName(String name) {
        this.name = name;
    }

    public synchronized String getEmail() {
        return email;
    }

    public synchronized void setEmail(String email) {
        this.email = email;
    }

    public synchronized boolean getSignatureUse() {
        return signatureUse;
    }

    public synchronized void setSignatureUse(boolean signatureUse) {
        this.signatureUse = signatureUse;
    }

    public synchronized String getSignature() {
        return signature;
    }

    public synchronized void setSignature(String signature) {
        this.signature = signature;
    }

    public synchronized String getDescription() {
        return description;
    }

    public synchronized void setDescription(String description) {
        this.description = description;
    }

    public synchronized String getReplyTo() {
        return replyTo;
    }

    public synchronized void setReplyTo(String replyTo) {
        this.replyTo = replyTo;
    }

    @Override
    public synchronized String toString() {
        return "Account.Identity(description=" + description + ", name=" + name + ", email=" + email + ", replyTo=" + replyTo + ", signature=" +
                signature;
    }
}

File: app/core/src/main/java/com/fsck/k9/Identity.kt
package com.fsck.k9

import android.os.Parcel
import android.os.Parcelable

data class Identity(
        val description: String? = null,
        val name: String? = null,
        val email: String? = null,
        val signature: String? = null,
        val signatureUse: Boolean = false,
        val replyTo: String? = null
) : Parcelable {
    override fun toString(): String {
        return "Account.Identity(description=" + description + ", name=" + name + ", email=" + email + ", replyTo=" + replyTo + ", signature=" +
                signature
    }

    // TODO remove when callers are converted to Kotlin
    fun withName(name: String?) = copy(name = name)
    fun withSignature(signature: String?) = copy(signature = signature)
    fun withSignatureUse(signatureUse: Boolean) = copy(signatureUse = signatureUse)
    fun withEmail(email: String?) = copy(email = email)

    constructor(parcel: Parcel) : this(
            parcel.readString(),
            parcel.readString(),
            parcel.readString(),
            parcel.readString(),
            parcel.readByte() != 0.toByte(),
            parcel.readString())

    override fun writeToParcel(parcel: Parcel, flags: Int) {
        parcel.writeString(description)
        parcel.writeString(name)
        parcel.writeString(email)
        parcel.writeString(signature)
        parcel.writeByte(if (signatureUse) 1 else 0)
        parcel.writeString(replyTo)
    }

    override fun describeContents(): Int {
        return 0
    }

    companion object CREATOR : Parcelable.Creator<Identity> {
        override fun createFromParcel(parcel: Parcel): Identity {
            return Identity(parcel)
        }

        override fun newArray(size: Int): Array<Identity?> {
            return arrayOfNulls(size)
        }
    }
}
